core/src/main/java/org/geysermc/geyser/translator/protocol/bedrock/entity/player/BedrockActionTranslator.java

```diff
@@ -324,35 +331,61 @@ public void translate(GeyserSession session, PlayerActionPacket packet) {
 
                     ServerboundUseItemOnPacket blockPacket = new ServerboundUseItemOnPacket(
-                            vector,
-                            Direction.DOWN,
-                            Hand.MAIN_HAND,
-                            0, 0, 0,
-                            false,
-                            session.getWorldCache().nextPredictionSequence());
+                        vector,
+                        Direction.DOWN,
+                        Hand.MAIN_HAND,
+                        0, 0, 0,
+                        false,
+                        session.getWorldCache().nextPredictionSequence());
                     session.sendDownstreamGamePacket(blockPacket);
                     break;
                 }
 
                 Entity itemFrame = ItemFrameEntity.getItemFrameEntity(session, packet.getBlockPosition());
                 if (itemFrame != null) {
                     ServerboundInteractPacket interactPacket = new ServerboundInteractPacket(itemFrame.getEntityId(),
-                            InteractAction.ATTACK, Hand.MAIN_HAND, session.isSneaking());
+                        InteractAction.ATTACK, Hand.MAIN_HAND, session.isSneaking());
                     session.sendDownstreamGamePacket(interactPacket);
                 }
-                break;
+            }
+        }
+    }
+
+    private void spawnBlockBreakParticles(GeyserSession session, Direction direction, Vector3i position, BlockState blockState) {
+        LevelEventPacket levelEventPacket = new LevelEventPacket();
+        switch (direction) {
+            case UP -> levelEventPacket.setType(LevelEvent.PARTICLE_BREAK_BLOCK_UP);
+            case DOWN -> levelEventPacket.setType(LevelEvent.PARTICLE_BREAK_BLOCK_DOWN);
+            case NORTH -> levelEventPacket.setType(LevelEvent.PARTICLE_BREAK_BLOCK_NORTH);
+            case EAST -> levelEventPacket.setType(LevelEvent.PARTICLE_BREAK_BLOCK_EAST);
+            case SOUTH -> levelEventPacket.setType(LevelEvent.PARTICLE_BREAK_BLOCK_SOUTH);
+            case WEST -> levelEventPacket.setType(LevelEvent.PARTICLE_BREAK_BLOCK_WEST);
         }
+        levelEventPacket.setPosition(position.toFloat());
+        levelEventPacket.setData(session.getBlockMappings().getBedrockBlock(blockState).getRuntimeId());
+        session.sendUpstreamPacket(levelEventPacket);
+    }
+
+    private void sendPlayerGlideState(GeyserSession session, Entity entity) {
+        ServerboundPlayerCommandPacket glidePacket = new ServerboundPlayerCommandPacket(entity.getEntityId(), PlayerState.START_ELYTRA_FLYING);
+        session.sendDownstreamGamePacket(glidePacket);
+    }
+
+    private void stopPlayerFlyingAbilities(GeyserSession session, Entity entity) {
+        // Otherwise gliding will not work in creative
+        ServerboundPlayerAbilitiesPacket playerAbilitiesPacket = new ServerboundPlayerAbilitiesPacket(false);
+        session.sendDownstreamGamePacket(playerAbilitiesPacket);
```

Comment:
this is okay to not be a method since its just used once

Reply:
Fixed in latest commit.